In [2]:
%load_ext sql

#### Connecting to the Database.

In [3]:
%%sql

sqlite:///CourseDataWarehouseDemo.db

'Connected: @CourseDataWarehouseDemo.db'

**We want to make sure that Anita's information is still correct from the data warehouse. She does have the same instructor ID from the database.**

In [5]:
%%sql

SELECT IID, Name FROM INSTRUCTOR_WH WHERE IID = 55;

 * sqlite:///CourseDataWarehouseDemo.db
Done.


[(55, 'Anita G.  Fernandez')]

**Quickly, we want to see the Title of some courses, and if there are any classes that have different locations and meeting times.**

In [43]:
%%sql

SELECT Title, LID, IID, MID
FROM COURSES_WH
WHERE IID = 55;

 * sqlite:///CourseDataWarehouseDemo.db
Done.


[('General Biology I (Majors)', 34, 55, 101),
 ('General Biology I (Majors)', 269, 55, 101),
 ('General Biology I (Majors)', 34, 55, 102),
 ('General Biology I (Majors)', 269, 55, 102),
 ('General Biology (Majors)', 29, 55, 101),
 ('General Biology (Majors)', 34, 55, 101),
 ('General Biology (Majors)', 269, 55, 101),
 ('General Biology I (Majors)', 34, 55, 102),
 ('General Biology I (Majors)', 269, 55, 102),
 ('Biology Undergraduate Teaching Practicum', 32, 55, 30),
 ('Biology Undergraduate Teaching Practicum', 274, 55, 30),
 ('Biology Teaching Practicum', 275, 55, 30),
 ('Biology Teaching Practicum', 276, 55, 30),
 ('Biology Teaching Practicum', 160, 55, 30),
 ('Biology Teaching Practicum', 277, 55, 30),
 ('Biology Teaching Practicum', 278, 55, 30),
 ('Internship-201409', 295, 55, 30),
 ('Internships 201409', 85, 55, 30),
 ('Internships 201409', 295, 55, 30),
 ('General Biology I', 34, 55, 1318),
 ('General Biology I', 8, 55, 1318),
 ('General Biology I', 34, 55, 1318),
 ('General Biology I', 34, 55, 1319),
 ('General Biology I', 34, 55, 1319),
 ('General Biology I', 54, 55, 1319),
 ('HonorsSem: Genetics,Ethics&Soc', 78, 55, 1803),
 ('HonorsSem: Genetics,Ethics&Soc', 92, 55, 1803),
 ('General Biology I Lab', 37, 55, 3790),
 ('General Biology I Lab', 59, 55, 3790),
 ('General Biology I Lab', 322, 55, 3790),
 ('Finding Answers: Genetics and Ethics', 61, 55, 5542),
 ('Finding Answers: Genetics and Ethics', 115, 55, 5542),
 ('Biology Undergraduate Teaching Practicum', 61, 55, 30),
 ('Biology Teaching Practicum', 65, 55, 30),
 ('Genetics', 53, 55, 6320),
 ('Genetics Lab', 45, 55, 6321),
 ('Genetics Lab', 112, 55, 6321),
 ('Genetics Lab', 45, 55, 6322),
 ('Genetics Lab', 66, 55, 6322),
 ('Genetics Lab', 45, 55, 7430),
 ('Genetics Lab', 48, 55, 7430),
 ('Developmental Biology', 19, 55, 7438),
 ('Developmental Biology', 43, 55, 7438),
 ('Developmental Biology Lab', 13, 55, 7439),
 ('Developmental Biology Lab', 44, 55, 7439),
 ('Genetics', 14, 55, 8542),
 ('Genetics Lab', 45, 55, 8544),
 ('Biology Research I', 96, 55, 8557),
 ('Biology Research I', 110, 55, 8557),
 ('Biology Research II', 32, 55, 8557),
 ('Biology Research II', 110, 55, 8557),
 ('Biology Research III', 110, 55, 8557),
 ('Biology Research IV', 110, 55, 8557),
 ('Genetics Lab', 45, 55, 9678),
 ('Developmental Biology', 41, 55, 9687),
 ('Developmental Biology Lab', 44, 55, 9688),
 ('Genetics', 14, 55, 10814),
 ('Genetics', 15, 55, 10814),
 ('Genetics Lab', 45, 55, 10815),
 ('Genetics Lab', 12, 55, 10816),
 ('Genetics Lab', 45, 55, 10816)]

**Our next check is to make sure that the query from our database works for Anita, and it does. Our data warehouse is telling us that Anita has taught 38 distinct classes.**

In [36]:
%%sql

SELECT COUNT (DISTINCT CRN) AS Classes FROM COURSES_WH WHERE IID = 55;

 * sqlite:///CourseDataWarehouseDemo.db
Done.


[(38,)]

**Theses are the amount of classes that Anita teaches based on the title of the class. Some of the titles may be slightly different, such as the 'Internship' title, as there is an 's' in one of them, and a '-' in the other. That seems to be a bug for the catalog/course, since they have the same serial number of 201409.**

In [40]:
%%sql

SELECT DISTINCT(Title), COUNT(Title) AS Count FROM COURSES_WH WHERE IID = 55 GROUP BY Title;

 * sqlite:///CourseDataWarehouseDemo.db
Done.


[('Biology Research I', 2),
 ('Biology Research II', 2),
 ('Biology Research III', 1),
 ('Biology Research IV', 1),
 ('Biology Teaching Practicum', 6),
 ('Biology Undergraduate Teaching Practicum', 3),
 ('Developmental Biology', 3),
 ('Developmental Biology Lab', 3),
 ('Finding Answers: Genetics and Ethics', 2),
 ('General Biology (Majors)', 3),
 ('General Biology I', 6),
 ('General Biology I (Majors)', 6),
 ('General Biology I Lab', 3),
 ('Genetics', 4),
 ('Genetics Lab', 11),
 ('HonorsSem: Genetics,Ethics&Soc', 2),
 ('Internship-201409', 1),
 ('Internships 201409', 2)]

**Now we want to see which locations Anita is teaching in. This _DOES NOT_ represent the total amount of class sessions that she has taught in like the database, but the amount of different locations each of her classes have been taught in.**

In [47]:
%%sql

SELECT LID, COUNT(*) AS Count
FROM COURSES_WH
WHERE IID = 55
GROUP BY LID
ORDER BY Count DESC;

 * sqlite:///CourseDataWarehouseDemo.db
Done.


[(34, 8),
 (45, 7),
 (110, 4),
 (269, 4),
 (14, 2),
 (32, 2),
 (44, 2),
 (61, 2),
 (295, 2),
 (8, 1),
 (12, 1),
 (13, 1),
 (15, 1),
 (19, 1),
 (29, 1),
 (37, 1),
 (41, 1),
 (43, 1),
 (48, 1),
 (53, 1),
 (54, 1),
 (59, 1),
 (65, 1),
 (66, 1),
 (78, 1),
 (85, 1),
 (92, 1),
 (96, 1),
 (112, 1),
 (115, 1),
 (160, 1),
 (274, 1),
 (275, 1),
 (276, 1),
 (277, 1),
 (278, 1),
 (322, 1)]

**If you are wondering where you might find Anita during her hours when she is teaching, your best bet is room 203 in the Nursing School, as she has had 8 classes in that room.**

In [48]:
%%sql

SELECT Location FROM LOCATION_WH WHERE LID = 34;

 * sqlite:///CourseDataWarehouseDemo.db
Done.


[('SON 203',)]